fahrschein/src/main/java/org/zalando/fahrschein/StreamParameters.java

```diff
@@ -59,22 +60,83 @@ String toQueryString() {
         return params.stream().collect(Collectors.joining("&"));
     }
 
-    public StreamParameters withBatchLimit(int batchLimit) {
+    /**
+     *
+     * Maximum number of Events in each chunk (and therefore per partition) of the stream.
+     *
+     * @param batchLimit
+     *          batch_limit must be lower or equal to stream_limit
+     * @return
+     * @throws StreamParametersException
+     */
+    public StreamParameters withBatchLimit(int batchLimit) throws StreamParametersException {
+        if(streamLimit != null && streamLimit < batchLimit){
+            throw new StreamParametersException("streamLimit is lower than batch_limit.");
+        }
         return new StreamParameters(batchLimit, streamLimit, batchFlushTimeout, streamTimeout, streamKeepAliveLimit, maxUncommittedEvents);
     }
 
-    public StreamParameters withStreamLimit(int streamLimit) {
+    /**
+     *
+     * Maximum number of Events in this stream (over all partitions being streamed in this connection).
+     *
+     * @param streamLimit
+     *          If 0 or undefined, will stream batches indefinitely.
```

Comment:
This is the issue I remembered, it seems this is not actually true, see https://github.com/zalando-incubator/nakadi-java/issues/125